### DISCUSSION TITLE
Documentation: [Self-hosted] Make Google SSO and Google Sheets datasource work together

### DISCUSSION THREAD

**Discussion by faiaz-halim** [ToolJet/ToolJet#9270] [General]
I hope someone finds this useful. 
If you are setting up a self-hosted tooljet instance with google sso you should be following this doc, https://docs.tooljet.com/docs/user-authentication/sso/google/
If you have already configured authorised redirect URL like ```https://<TOOLJET_HOST>/sso/google/<workspace_id>``` or just simple ```https://<TOOLJET_HOST>/sso/google/```, it will work for sso login with google.
If you need to configure google sheets as data source you should be following this doc, https://docs.tooljet.com/docs/data-sources/google.sheets/
But you need to add another redirect uri to your credential in google cloud console which looks like this ```https://<TOOLJET_HOST>/oauth2/authorize``` otherwise the google sheets datasource will not be able to connect.
Maybe the doc https://docs.tooljet.com/docs/data-sources/google.sheets/ needs an update?

**Comment by karan-rathod-316** [comment]
Hi @faiaz-halim, we are taking a look at this issue. We will test it and update the docs if needed. Thank you for providing the context in detail:-).
